File: projects/2023/project08_highlevel_languages/PythonCode/src/input_loader.py
import numpy as np
from pandas import read_csv

def load_input():
    # this turns alpha into int(alpha) which is 0
    # return np.genfromtxt('../../universal_input/input_dimensions.csv', delimiter=',', skip_header=1, dtype=np.int32)
    data = read_csv("../../universal_input/input_dimensions.csv")

    # Extract and store rows as tuples
    rows_as_tuples = []
    for index, row in data.iterrows():
        row_tuple = (int(row['x_dim']), int(row['y_dim']), int(row['z_dim']), float(row['alpha']), int(row['num_iter']))
        rows_as_tuples.append(row_tuple)

    return rows_as_tuples

def generate_initial_array(dim_df):

    num_halo = 2
    nx = dim_df[0]
    ny = dim_df[1]
    nz = dim_df[2]

    arr = np.zeros((nz, ny + 2 * num_halo, nx + 2 * num_halo))
    arr[
        nz // 4 : 3 * nz // 4,
        num_halo + ny // 4 : num_halo + 3 * ny // 4,
        num_halo + nx // 4 : num_halo + 3 * nx // 4,
    ] = 1.0

    return arr

File: projects/2023/project08_highlevel_languages/JuliaCode/src/input_loader.jl
module InputLoader

using CSV, DataFrames

export load_input, generate_initial_array

function load_input()
    return CSV.read("./universal_input/input_dimensions.csv", DataFrame, header=true)
end

function generate_initial_array(dim_df, num_halo = 2)
    nx = floor(Int,dim_df[1] + 2 * num_halo)
    ny = floor(Int,dim_df[2] + 2 * num_halo)
    nz = floor(Int,dim_df[3])

    arr = zeros(Float64, nx, ny, nz)
    arr[(1 + nx ÷ 4):(3 * nx ÷ 4), (1 + ny ÷ 4):(3 * ny ÷ 4), (1 + nz ÷ 4):(3 * nz ÷ 4)] .= 1.0
    return arr
end

end #module